lib/src/main/java/com/diffplug/spotless/kotlin/DiKTatStep.java

```diff
@@ -0,0 +1,168 @@
+
+			HashSet<String> pkgSet = new HashSet<>();
+			pkgSet.add(MAVEN_COORDINATE + versionDiktat);
+
+			this.userData = new TreeMap<>(userData);
+			this.pkg = PACKAGE_DIKTAT;
+			this.pkgKtlint = PACKAGE_KTLINT;
+			this.jar = JarState.from(pkgSet, provisioner);
+			this.isScript = isScript;
+			this.configPath = configPath;
+		}
+
+		FormatterFunc createFormat() throws Exception {
+
+			ClassLoader classLoader = jar.getClassLoader();
+
+			// first, we get the diktat rules
+			if (configPath != null) {
+				System.setProperty("diktat.config.path", configPath);
+			}
+
+			Class<?> ruleSetProviderClass = classLoader.loadClass(pkg + ".ruleset.rules.DiktatRuleSetProvider");
+			Object standardRuleSet = ruleSetProviderClass.getMethod("get").invoke(ruleSetProviderClass.newInstance());
+			Iterable<?> ruleSets = Collections.singletonList(standardRuleSet);
+
+			// next, we create an error callback which throws an assertion error when the format is bad
+			Class<?> function2Interface = classLoader.loadClass("kotlin.jvm.functions.Function2");
+			Class<?> lintErrorClass = classLoader.loadClass(pkgKtlint + ".ktlint.core.LintError");
+			Method detailGetter = lintErrorClass.getMethod("getDetail");
+			Method lineGetter = lintErrorClass.getMethod("getLine");
+			Method colGetter = lintErrorClass.getMethod("getCol");
+			Object formatterCallback = Proxy.newProxyInstance(classLoader, new Class[]{function2Interface},
+				(proxy, method, args) -> {
+					Object lintError = args[0]; //ktlint.core.LintError
+					boolean corrected = (Boolean) args[1];
+					if (!corrected) {
+						String detail = (String) detailGetter.invoke(lintError);
+						int line = (Integer) lineGetter.invoke(lintError);
+						int col = (Integer) colGetter.invoke(lintError);
+						throw new AssertionError("Error on line: " + line + ", column: " + col + "\n" + detail);
+					}
+					return null;
+				});
+
+			// grab the KtLint singleton
+			Class<?> ktlintClass = classLoader.loadClass(pkgKtlint + ".ktlint.core.KtLint");
+			Object ktlint = ktlintClass.getDeclaredField("INSTANCE").get(null);
+			FormatterFunc.NeedsFile formatterFunc;
+			// and its format method
+
+			Class<?> paramsClass = classLoader.loadClass(pkgKtlint + ".ktlint.core.KtLint$Params");
+			// and its constructor
+			Constructor<?> constructor = paramsClass.getConstructor(
+				/* fileName, nullable */ String.class,
+				/* text */ String.class,
+				/* ruleSets */ Iterable.class,
+				/* userData */ Map.class,
+				/* callback */ function2Interface,
+				/* script */ boolean.class,
+				/* editorConfigPath, nullable */ String.class,
+				/* debug */ boolean.class);
+			Method formatterMethod = ktlintClass.getMethod("format", paramsClass);
+			formatterFunc = (input, file) -> {
+				userData.put("file_path", file.getAbsolutePath());
+				try {
+					Object params = constructor.newInstance(
+						/* fileName, nullable */ null,
```

Comment:
```suggestion
						/* fileName, nullable */ file.name,
```
Because I believe we need this